File: client/src/app/core/services/project.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { API_BASE_URL } from '../config/api.config';
import { ProjectCreateRequest, ProjectResponse, ProjectUpdateRequest } from '../models/project.model';

const BASE = `${API_BASE_URL}/api/projects`;

@Injectable({ providedIn: 'root' })
export class ProjectService {
  private readonly http = inject(HttpClient);

  create(request: ProjectCreateRequest): Observable<ProjectResponse> {
    return this.http.post<ProjectResponse>(BASE, request);
  }

  list(): Observable<ProjectResponse[]> {
    return this.http.get<ProjectResponse[]>(BASE);
  }

  get(id: string): Observable<ProjectResponse> {
    return this.http.get<ProjectResponse>(`${BASE}/${id}`);
  }

  update(id: string, request: ProjectUpdateRequest): Observable<ProjectResponse> {
    return this.http.patch<ProjectResponse>(`${BASE}/${id}`, request);
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${BASE}/${id}`);
  }
}

File: Customer/src/main/java/com/postman/alt/service/ProjectService.java
package com.postman.alt.service;

import com.postman.alt.service.dto.ProjectCreateRequest;
import com.postman.alt.service.dto.ProjectResponse;
import com.postman.alt.service.dto.ProjectUpdateRequest;

import java.util.List;
import java.util.UUID;

public interface ProjectService {

    ProjectResponse create(UUID requesterId, ProjectCreateRequest request);

    List<ProjectResponse> listForOrg(UUID requesterId);

    ProjectResponse get(UUID projectId, UUID requesterId);

    ProjectResponse update(UUID projectId, UUID requesterId, ProjectUpdateRequest request);

    void delete(UUID projectId, UUID requesterId);
}
